### PR TITLE
Revised download() so parallel get() is utilized. Small refactor.

### PR DESCRIPTION
**Pull request by kxk302**
(Please replace this header with a description of your pull request. Please include *BOTH* what you did and why you made the changes. The "why" may simply be citing a relevant Galaxy issue.)
(If fixing a bug, please add any relevant error or traceback)
(For UI components, it is recommended to include screenshots or screencasts)

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: lib/galaxy/objectstore/irods.py [hunk 1]
```diff
@@ -350,27 +350,16 @@ def _download(self, rel_path):
         data_obj = None
 
         try:
-            data_obj = self.session.data_objects.get(data_object_path)
+            cache_path = self._get_cache_path(rel_path)
```
Comment (kxk302) [inline lib/galaxy/objectstore/irods.py:351]: Must pass the path to get(), so get() can download the file utilizing multiple threads, if necessary. Getting data object instance, reading and writing it to cache will not utilize multithreading and is removed.

File: lib/galaxy/objectstore/irods.py [hunk 2]
```diff
@@ -406,10 +395,10 @@ def _push_to_irods(self, rel_path, source_file=None, from_string=None):
             # Create sub-collection first
             self.session.collections.create(collection_path, recurse=True)
 
-            # Create data object
-            data_obj = self.session.data_objects.create(data_object_path, self.resource, **options)
-
             if from_string:
+                # Create data object
```
Comment (kxk302) [inline lib/galaxy/objectstore/irods.py:397]: This is needed only in this path through the code. 
Also, I do not see any callers, where 'from_string' is set. Can this if clause be removed?

**Comment by kxk302** [comment]
@mvdbeek Could you please take a look at this PR tomorrow? Thanks!

File: lib/galaxy/objectstore/irods.py (hunk 2 above)
Comment (mvdbeek) [reply lib/galaxy/objectstore/irods.py:397]: Yes, you can remove it.

**Comment by kxk302** [comment]
Thx @mvdbeek!

**Pull request merged**